File: examples/picoenvmon/java/picoenvmon/EnvModule.java
// SPDX-License-Identifier: GPL-3.0-only
package picoenvmon;

import javax.inject.Singleton;
import picodroid.content.SharedPreferences;
import picodroid.di.Module;
import picodroid.di.Provides;

/**
 * Bindings for SDK types, which cannot carry an {@code @Inject} constructor. Installed
 * automatically (single implicit component).
 */
@Module
public final class EnvModule {
  private EnvModule() {}

  /** The app's preferences file, opened once. */
  @Provides
  @Singleton
  static SharedPreferences providePrefs() {
    return SharedPreferences.open(EnvApp.PREFS_NAME);
  }
}

File: examples/picoenvmon_kt/kotlin/picoenvmonkt/EnvModule.kt
// SPDX-License-Identifier: GPL-3.0-only
package picoenvmonkt

import javax.inject.Singleton
import picodroid.content.SharedPreferences
import picodroid.di.Module
import picodroid.di.Provides

/**
 * Bindings for SDK types, which cannot carry an `@Inject` constructor. Installed automatically
 * (single implicit component). A `@Module object` with `@JvmStatic` providers is the Kotlin form of
 * Java's `final class` + `static` method: the processor calls it statically and never needs an
 * instance.
 */
@Module
object EnvModule {
    /** The app's preferences file, opened once. */
    @Provides
    @Singleton
    @JvmStatic
    fun providePrefs(): SharedPreferences = SharedPreferences.open(PREFS_NAME)
}
